test/lazy/test_ddp.py

```diff
@@ -0,0 +1,149 @@
+# Owner(s): ["oncall: jit"]
+
+import copy
+import os
+import sys
+import torch
+import torch._lazy
+import torch._lazy.ts_backend
+import torch.distributed as dist
+import torch.nn as nn
+import torch.optim as optim
+
+from datetime import timedelta
+from torch._utils_internal import TEST_MASTER_ADDR as MASTER_ADDR
+from torch._utils_internal import TEST_MASTER_PORT as MASTER_PORT
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.testing._internal.common_utils import run_tests, TEST_WITH_DEV_DBG_ASAN, NO_MULTIPROCESSING_SPAWN
+from torch.testing._internal.common_distributed import MultiProcessTestCase
+
+if not dist.is_available():
+    print("Distributed not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+if not torch.cuda.is_available() or not dist.is_nccl_available():
+    print("CUDA/NCCL not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+if TEST_WITH_DEV_DBG_ASAN:
+    print("Skip dev-asan as torch + multiprocessing spawn have known issues", file=sys.stderr)
+    sys.exit(0)
+
+if NO_MULTIPROCESSING_SPAWN:
+    print("Spawn not available, skipping tests.", file=sys.stderr)
+    sys.exit(0)
+
+torch._lazy.ts_backend.init()
+os.environ["WORLD_SIZE"] = str(torch.cuda.device_count())
+
+class ToyModel(nn.Module):
+    def __init__(self):
+        super(ToyModel, self).__init__()
+        self.net1 = nn.Linear(10, 10)
+        self.relu = nn.ReLU()
+        self.net2 = nn.Linear(10, 5)
+
+    def forward(self, x):
+        return self.net2(self.relu(self.net1(x)))
+
+
+def step(model, input, labels, device, loss_fn, optimizer):
+    input = input.to(device)
+    labels = labels.to(device)
+    optimizer.zero_grad()
+
+    output = model(input)
+
+    loss = loss_fn(output, labels)
+    loss.backward()
+    optimizer.step()
+    # FIXME(alanwaketan): Investigate why we produce different
+    # results if mark_step.
+    # if device.type == "lazy":
+    #     torch._lazy.mark_step(str(device))
+    return loss
+
+def init(model, device, rank, size):
+    model = copy.deepcopy(model).to(device)
+    if device.type == "lazy":
+        # Somehow, LazyTensor crashes if gradient_as_bucket_view is not set.
+        # FIXME(alanwaketan): Investigate why and if we can remove this constraint.
+        model = DDP(model, gradient_as_bucket_view=True)
+        # FIXME(alanwaketan): Do we need this or not?
```

Comment:
did you conclude that we do not need this?

Reply:
This was originally added for the mark_step to avoid NaN loss issue. So I need to investigate mark_step issue to further determine.

Reply:
ok, so you're investigating now? or you are planning to land first?

Reply:
Can I land this first and then investigate as a follow up?

Reply:
my general rule of thumb is that you can land with a bug/todo if you know what/why it is happening and make a good argument that it's not a big factor for users.  Hence i'd be a little concerned with 'investigating as a follow up'.  It may not be a big deal, but i'm looking to you to justify that before landing

Reply:
The graphs produced with mark_step is vastly different than the one reproduced without mark_step. Looking into why.